### DISCUSSION TITLE
Getting Preview URLs of sibling service (monorepo)

### DISCUSSION THREAD

**Discussion by cyrus-za** [vercel/vercel#5231] [Help] [answered]
We got a monorepo with backend and frontend services all deployed as their own vercel project. The `preview urls` obviously get their ENV from the env setup for preview, but those are hardcoded. We want to programatically identify the url for the backend service to point to from the frontend. I know the github branch name can be obtained, but how do I get the url based on that, so that any given preview url for a frontend (nextjs) app points to the same version/branch of a backend app?

**Comment by leo** [comment, marked as answer]
Thanks for raising this feedback!

We currently don't provide a feature for this, but I added this to our internal roadmap, as it's definitely something we should add in the future.

Once it's been added, we'll update you on this thread right here.

**Reply by cyrus-za** [reply]
Great thanks. I look forward to it

**Reply by rushilsrivastava** [reply]
Any update for this?

**Reply by jesse-holden** [reply]
Would love for this feature to be added to Vercel, we have a use case where we need to create Next JS rewrites at build time to point to sibling preview deployments. It would be nice to have an official way to do this, instead of hacking together a substring replacement function to guess the preview URL of the other deployed packages.

**Reply by miquelvir** [reply]
Same!

**Reply by will-holley** [reply]
+1

**Reply by lsemerini** [reply]
I am currently having to do something like replacing the predictable part of `VERCEL_BRANCH_URL` it's really hacky. Is there a more elegant way to handle this in the meantime?

**Reply by Olex93** [reply]
+1

**Reply by danteissaias** [reply]
Any progress on this?
